spec/unit/pdk/cli/build_spec.rb

```diff
@@ -2,18 +2,30 @@
 
 describe 'PDK::CLI build' do
   let(:help_text) { a_string_matching(%r{^USAGE\s+pdk build}m) }
+  let(:command_opts) { [] }
+
+  shared_context 'exits cleanly' do
+    after(:each) do
+      PDK::CLI.run(['build'] + command_opts)
+    end
+  end
+
+  shared_context 'exits with an error' do |exit_code|
+    after(:each) do
+      expect {
+        PDK::CLI.run(['build'] + command_opts)
+      }.to raise_error(SystemExit) { |error|
+        expect(error.status).to eq(exit_code)
+      }
+    end
+  end
```

Comment:
The shared contexts here were merely so I didn't have to copy and paste the `after(:each)` block into every context that has needs to check for their respective exit. I could put the `--force` tests into their own top-level context, but I thought it would be more maintainable to keep the contexts associated with the top-level usecases like `incomplete metadata`  or `complete metadata` but `not pdk compatible`. A failure of the expectation will look like any expectation failure, it will show that the exit code doesn't match the expected exit code.

What do you think @rodjek ?

Reply:
Maybe a shared example would address both of our concerns? It just seems like semantically a shared "context" is for multiple tests that need the same pre-conditions in place before the test runs.

Reply:
In terms of cleanliness, I'd prefer to see it implemented as a custom matcher, purely because I don't like to see `expect` used outside of example `it` blocks (`before` and `after` should only really be used for test setup & tear down)